components/local-app/main.go

```diff
@@ -261,6 +261,10 @@ func tryConnectToServer(gitpodUrl string, tkn string, reconnectionHandler func()
 		Log:                 logrus.NewEntry(logrus.StandardLogger()),
 		ReconnectionHandler: reconnectionHandler,
 		CloseHandler:        closeHandler,
+		ExtraHeaders: map[string]string{
+			"User-Agent":       "gitpod/local-companion",
+			"X-Client-Version": Version,
```

Comment:
@geropl Can we add user agent, and version properties to tracing information? Should we do it another PR?

Reply:
> an we add user agent, and version properties to tracing information?

The first step is to send version/client info on each connection. I will create a PR to add this to `server` traces.

To expand on this, it would be awesome to tracing in `supervisor` as well, including the user-agent and version info. Also, it would be awesome if we could send a `traceId` from `supervisor` to `server` on each API call (as opposed to per-connection like user-agent, version). But I'm not sure if there's sth like per-calll metadata in JsonRPC that we could leverage. Or build ourselves. :slightly_smiling_face:

Reply:
@iQQBot Could you check please whether we have version in supervisor to provide it as `X-Client-Version` there? I think each Gitpod component should have some version file.

Reply:
> Also, it would be awesome if we could send a traceId from supervisor to server on each API call (as opposed to per-connection like user-agent, version). But I'm not sure if there's sth like per-calll metadata in JsonRPC that we could leverage. Or build ourselves.

We could change transport to add it to each json-rpc message on golang and typescript side. What would be a trace id?

Reply:
Cmp: https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/leeway.Dockerfile#L25-L26

Reply:
Will adding something like help:
```
ENV GITPOD_ENV_SET_SUPERVISOR_COMMIT=${__GIT_COMMIT}
ENV GITPOD_ENV_SET_SUPERVISOR_VERSION=${VERSION}
```
`GITPOD_ENV_SET` is a registry-facade directive to set env vars: https://github.com/gitpod-io/gitpod/blob/56fe86ff90a5c0bb652d2ab8a41bb22545a5b34c/components/registry-facade/pkg/registry/layersource.go#L220

We could extend [WorkspaceConfiig](https://github.com/gitpod-io/gitpod/blob/d70eb530702440ac8c0eb91dd30270eebe7e9c1d/components/supervisor/pkg/supervisor/config.go#L178) with it?

Reply:
sounds good if it works

Reply:
@geropl For trace if on each request I think it is better to deal with another PR. Will you file an issue?

Reply:
> We could change transport to add it to each json-rpc message on golang and typescript side. What would be a trace id?

That sounds good!

> For trace if on each request I think it is better to deal with another PR. Will you file an issue?

Definitely, and will.

@akosyakov